src/Build.UnitTests/Evaluation/Evaluator_Tests.cs

```diff
@@ -914,6 +914,44 @@ public void MultipleImportsVerifyImportsIncludingDuplicates()
             }
         }
 
+        [DotNetOnlyFact("Tests .NET SDK-only error")]
+        public void ImportWithVSPathThrowsCorrectError()
+        {
+            InvalidProjectFileException ex = Assert.Throws<InvalidProjectFileException>(() =>
+            {
+                string projectPath = null;
+                string importPath = null;
+
+                try
+                {
+                    // Does not matter that the file or folder does not exist, we are checking for the VS pathing here
+                    importPath = "path\\that\\does\\not\\exist\\Microsoft\\VisualStudio\\FileName.txt";
+                    projectPath = FileUtilities.GetTemporaryFileName();
+
+                    string import = ObjectModelHelpers.CleanupFileContents(@"
+                            <Project ToolsVersion=""msbuilddefaulttoolsversion"" xmlns='msbuildnamespace' >
+                            </Project>
+                        ");
+
+                    File.WriteAllText(projectPath, import);
+
+                    string content = ObjectModelHelpers.CleanupFileContents(@"
+                            <Project ToolsVersion=""msbuilddefaulttoolsversion"" xmlns='msbuildnamespace' >
+                                <Import Project='" + importPath + @"'/>
+                            </Project>
+                        ");
```

Comment:
For new and updated test code, try to prefer using `TestEnvironment` to write files (it handles cleanup too).